Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorPaintEffect.cxx

```diff
@@ -1558,49 +1536,40 @@ void qSlicerSegmentEditorPaintEffect::updateGUIFromMRML()
   bool brushDiameterIsRelative = this->integerParameter("BrushDiameterIsRelative");
 
   d->BrushDiameterRelativeToggle->blockSignals(true);
-  d->BrushDiameterRelativeToggle->setText(brushDiameterIsRelative ? "%" : tr("mm"));
+  d->BrushDiameterRelativeToggle->setChecked(!brushDiameterIsRelative);
   d->BrushDiameterRelativeToggle->blockSignals(false);
 
-  d->BrushDiameterSlider->blockSignals(true);
+  d->BrushDiameterSliderWidget->blockSignals(true);
   if (brushDiameterIsRelative)
     {
-    d->BrushDiameterSlider->setMinimum(1);
-    d->BrushDiameterSlider->setMaximum(25);
-    d->BrushDiameterSlider->setValue(this->doubleParameter("BrushRelativeDiameter"));
-    d->BrushDiameterSlider->setSingleStep(1);
+    d->BrushDiameterSliderWidget->setRange(1, 25);
+    d->BrushDiameterSliderWidget->setValue(this->doubleParameter("BrushRelativeDiameter"));
+    d->BrushDiameterSliderWidget->setSingleStep(1);
     }
   else
     {
-    d->BrushDiameterSlider->setMinimum(this->doubleParameter("BrushMinimumAbsoluteDiameter"));
-    d->BrushDiameterSlider->setMaximum(this->doubleParameter("BrushMaximumAbsoluteDiameter"));
-    d->BrushDiameterSlider->setValue(this->doubleParameter("BrushAbsoluteDiameter"));
-    d->BrushDiameterSlider->setSingleStep(this->doubleParameter("BrushMinimumAbsoluteDiameter"));
+    d->BrushDiameterSliderWidget->setRange(this->doubleParameter("BrushMinimumAbsoluteDiameter"), this->doubleParameter("BrushMaximumAbsoluteDiameter"));
+    d->BrushDiameterSliderWidget->setValue(this->doubleParameter("BrushAbsoluteDiameter"));
+    d->BrushDiameterSliderWidget->setSingleStep(this->doubleParameter("BrushMinimumAbsoluteDiameter"));
     }
-  d->BrushDiameterSlider->blockSignals(false);
-
-
-  d->BrushDiameterSpinBox->blockSignals(true);
-  d->BrushDiameterSpinBox->setMRMLScene(this->scene());
-  d->BrushDiameterSpinBox->setMinimum(d->BrushDiameterSlider->minimum());
-  d->BrushDiameterSpinBox->setMaximum(d->BrushDiameterSlider->maximum());
-  d->BrushDiameterSpinBox->setValue(d->BrushDiameterSlider->value());
   if (brushDiameterIsRelative)
     {
-    d->BrushDiameterSpinBox->setQuantity("");
-    d->BrushDiameterSpinBox->setSuffix("%");
-    d->BrushDiameterSpinBox->setDecimals(0);
+    d->BrushDiameterSliderWidget->setQuantity("");
+    d->BrushDiameterSliderWidget->setSuffix("%");
+    d->BrushDiameterSliderWidget->setDecimals(0);
     }
   else
     {
-    d->BrushDiameterSpinBox->setQuantity("length");
-    d->BrushDiameterSpinBox->setUnitAwareProperties(qMRMLSpinBox::Prefix | qMRMLSpinBox::Suffix);
-    int decimals = (int)(log10(d->BrushDiameterSlider->minimum()));
+    d->BrushDiameterSliderWidget->setQuantity("length");
+    d->BrushDiameterSliderWidget->setSuffix(tr("mm"));
```

Comment:
This is not necessary. If the MRML scene is set in the widget, suffix is set to be unit-aware, and quantity is set to `length` then the widget gets the current suffix from the scene.

Reply:
Ok I can remove these. This wasn't new code as the `BrushDiameterSpinBox`, a qMRMLSpinBox, was previously having the quantity and suffix properties being set. See 
https://github.com/Slicer/Slicer/blob/3f425197ee857eb1ebffb13f9c24bdc3b0066b69/Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorPaintEffect.cxx#L1589-L1590

Reply:
Suffix need to be set for %, but not for physical unit (then the widget sets for itself).

Reply:
I've pushed to remove the following lines
```c++
    d->BrushDiameterSliderWidget->setQuantity("length");
    d->BrushDiameterSliderWidget->setSuffix(tr("mm"));
```